main/jlp/widgets/widget_factory.cpp

```
@@ -2595,6 +2613,288 @@ lv_obj_t* build_voice(BuildCtx& ctx, JsonObjectConst spec, std::string* err) {
+  lv_indev_t* indev = lv_indev_active();
+  if (!indev) return;
+  lv_point_t p;
+  lv_indev_get_point(indev, &p);
+  if (type == 1 && p.x == c->last_sent.x && p.y == c->last_sent.y) return;
+  c->last_sent = p;
+
+  // Logical (rotated) screen coords match the captured frame 1:1 as long as
+  // capture resolution == panel resolution; only the widget offset differs.
+  lv_area_t a;
+  lv_obj_get_coords(static_cast<lv_obj_t*>(lv_event_get_target(e)), &a);
+  int32_t x = p.x - a.x1;
```

Clang (14.0.6)

[warning] variable name 'x' is too short, expected at least 3 characters

(readability-identifier-length)